configure.ac

```diff
@@ -3168,20 +3168,26 @@ XINPUT_STATE_EX s1;
 dnl Check for the dlfcn.h interface for dynamically loading objects
 CheckDLOPEN()
 {
+dnl NOTE: CheckDLOPEN is called only for relevant platforms.
+    AC_DEFINE(DYNAPI_NEEDS_DLOPEN, 1, [ ])
+
+    AC_MSG_CHECKING(for dlopen)
```

Comment:
Moving this test out of the sdl-dlopen block seems incorrect. What was the reason for that?

Reply:
For the record, this is very incomplete at the moment - needs more work.

As for the reason: 'allow'ing dlopen seems weird: all others enable/disable loadso
functionality and not its backend, hence the move. Should I not do that?

Reply:
That option is really old and I no longer remember why we needed it. It's definitely 15+ years ago and may no longer be relevant. Moving it out and just checking to see if it's available seems sufficient, and we can remove the sdl-dlopen option.

If this isn't ready to merge, can you mark it as a draft in GitHub's interface, so I don't accidentally try to merge it before you're ready?

Reply:
Yeah, marked it as draft for the moment. Will get back to it (might not make
it to 2.0.16, whenever that's going to happen.)